spec/jobs/invoices/abacus/create_yearly_abo_alpen_invoices_job_spec.rb

```diff
@@ -0,0 +1,378 @@
+#  hitobito_sac_cas and licensed under the Affero General Public License version 3
+#  or later. See the COPYING file at the top-level directory or at
+#  https://github.com/hitobito/hitobito_sac_cas
+
+require "spec_helper"
+
+describe Invoices::Abacus::CreateYearlyAboAlpenInvoicesJob do
+  let(:subject) { described_class.new }
+  let(:group) { groups(:abo_die_alpen) }
+  let(:exception) { StandardError.new("Something went wrong") }
+
+  let!(:abonnent) { people(:abonnent) }
+  let!(:abonnent_role) { roles(:abonnent_alpen) }
+
+  let!(:abonnent_2) { Fabricate(:person, abacus_subject_key: "124", country: "CH") }
+  let!(:abonnent_role_2) { create_role(abonnent_2) }
+
+  let!(:abonnent_3) { Fabricate(:person, abacus_subject_key: "125", country: "DE", zip_code: "12345") }
+  let!(:abonnent_role_3) { create_role(abonnent_3) }
+
+  let!(:abonnent_4) { Fabricate(:person, abacus_subject_key: "126", country: "CH") }
+  let!(:abonnent_role_4) { create_role(abonnent_4) }
+
+  let!(:abonnent_5) { Fabricate(:person, abacus_subject_key: "127", country: "CH", language: "fr") }
+  let!(:abonnent_role_5) { create_role(abonnent_5) }
+
+  let!(:abonnent_6) { Fabricate(:person, abacus_subject_key: "128", country: "CH") }
+  let!(:abonnent_role_6) { create_role(abonnent_6) }
+
+  def create_role(person)
+    Fabricate(:role, type: Group::AboMagazin::Abonnent.sti_name,
+      person: person,
+      group: group,
+      start_on: 10.days.ago,
+      end_on: 50.days.from_now)
+  end
+
+  before do
+    Group.root.update!(abo_alpen_fee: 60, abo_alpen_postage_abroad: 16)
+    abonnent.update_columns(abacus_subject_key: "123", country: "CH")
+    abonnent_role.update_column(:end_on, 50.days.from_now)
+  end
+
+  describe "#error" do
+    it "creates log entry when job crashes" do
+      expect { subject.error(subject, exception) }.to change { HitobitoLogEntry.where(level: :error).count }.by(1)
+
+      expect(HitobitoLogEntry.last.message).to eq "Jahresrechnungen Abo Magazin Die Alpen konnten nicht an Abacus übermittelt werden. " \
+              "Es erfolgt ein weiterer Versuch."
+    end
+  end
+
+  describe "#failure" do
+    it "creates log entry when job fails" do
+      expect { subject.failure(subject) }.to change { HitobitoLogEntry.where(level: :error).count }.by(1)
+
+      expect(HitobitoLogEntry.last.message).to eq "Rollierendes Inkassolauf Abo Magazin Die Alpen abgebrochen."
+    end
+  end
+
+  describe "#active_abonnenten" do
+    it "includes every abonnent role where role end in next 62 days" do
+      expect(subject.active_abonnenten).to match_array [abonnent_role, abonnent_role_2, abonnent_role_3, abonnent_role_4, abonnent_role_5, abonnent_role_6]
+    end
+
+    it "does no unclude abonnent where role ends in more than 62 days" do
+      abonnent_role.update_column(:end_on, 70.days.from_now)
+      expect(subject.active_abonnenten).to match_array [abonnent_role_2, abonnent_role_3, abonnent_role_4, abonnent_role_5, abonnent_role_6]
```

Comment:
Vielleicht zusätzlich noch ein `expect(...).to_not include(...)` dann stimmts noch mit der `it` Beschreibung überein und ist imo besser lesbar